src/common.cu

```diff
@@ -80,6 +83,125 @@ static int cudaGraphLaunches = 0;
 static int report_cputime = 0;
 // Report average iteration time: (0=RANK0,1=AVG,2=MIN,3=MAX)
 static int average = 1;
+static std::string resultsFile;
+
+//TODO: implement metaInfo method to peint metadata (device info etc.)
+Reporter::Reporter(std::string csvName_, const char* timeStr_, bool isMain_): csvName { csvName_ }, timeStr {timeStr_}, isMainThreadFlag { isMain_ } {
+  if (!isMainThread()){
+    return;
+  }
+  persistanceMode = csvName != "";
```

Comment:
I'd rather call everything `outputMode` rather than `persistanceMode`. Not sure "persistance" would be well understood.